### ISSUE TITLE
Get SKU as MPN in the product feed ?

### ISSUE THREAD

**Issue by donmen2** [facebookincubator/facebook-for-magento#36]
Hi,

I need to get (the same way and position in the feed as gender) the SKU of each product, at the end of each line. If you like, I would like to get the SKU instead of the gender... just to be named in the column name MPN...

How can I do that?

**Comment by mengyingdu** [comment]
Hi @donmen2 ,

Our feed generation is fixed format in the code, i'm not sure if you change it, it can seamlessly work with the uploading.

Can you explain why you want this? Maybe we could have some other ways.

Thanks.

**Comment by donmen2** [comment]
Hi,
I need the sku from my magento website for each feed line, at the very end. This way, the Facebook ads will show a user that visited my X product, when he/she goes to Facebook, the exact product he/she visited on my website. Otherwise, the ads on facebook will be shown randomly, without any result.

In the attachment you can see the required fields for a Facebook Data Feed, as required on this page: https://www.facebook.com/business/help/1397294963910848

Basically, instead of the "gender" field, I need to replace it by the SKU. The column header needs to be called "mpn"... That is basically all...

Thank you very much in advance... You are vey kind!

![img_3903](https://user-images.githubusercontent.com/36992206/36915214-91ab1a70-1e58-11e8-91e0-031543373b25.JPG)

**Comment by mengyingdu** [comment]
Hi @donmen2,
There should be a column named 'brand' in your feed. Isn't it enough?
Thanks.

**Comment by mengyingdu** [comment]
Hi @donmen2 ,

I think magento doesn't support mpn attribute in their stock data.
https://stackoverflow.com/questions/6496985/magento-fetching-mpn-manufacturer-part-no-or-upc-universal-product-code
Unfortunately, if magento doesn't support it themselves, it's impossible to support it on our site since we need their API.

Thanks.

**Comment by donmen2** [comment]
I know that magento has no mpn feature but I need each product's sku (the brand name is not enough for matching the product visited on my website to the one advertised on facebook.) that can only be done through that sku, which is unique for each product. I only need to name the column MPN, but it would actually contain.the sku values. My website has only one brand, our own brand name, so that wouldn't help.at all. Is there a way to get the sku in the feed?

**Comment by donmen2** [comment]
Is it possible to get the sku for each product using your extension?

**Comment by mengyingdu** [comment]
Hi @donmen2 ,

The product data our plugin is querying your db by magento API, i don't think it's possible to skip this unsupported feature. 

As for your intention for advertising, if you check you catalog, each product has its FBID, which will be used in your advertisement.

For example, you have a product set, probably one product get more visit. If you set up a dynamic catalog ads, our ranking system will automatically optimized the Ads for you. Does it fulfill your questions?

Thanks.

**Comment by donmen2** [comment]
Hi @mengyingdu,

Thank you for answering me. Unfortunately it is not enough. For the advertisements to be properly displayed for each customer and increase the chances of a sale, I really need the SKU of each product... It is not asked for by facebook in a datafeed for nothing... they use it to make a connection between the product somebody visited on my website and the advertisement facebook should show to that customer while browsing Facebook...

The FBID doesn't really help that much. It is natural to have a FBID for each product so they can catalogue your products, but that doesn't solve the problem of a quality matching...

Imagine somebody visits my website and goes to a skirt. What if that person goes to facebook and facebook shows her a pair of snickers? She won't be interested, as she manifested the interest for a skirt on my website, not for a pair of snickers... This is where that sku comes on the stage... It helps facebook deliver the right ad to the right person.

Thank you... I will look for other extension if this one can't do that. Thank you very much

**Comment by mengyingdu** [comment]
Hi @dmitridr ,

Can you help double check whether it is possible? I have initial research as i mentioned in the reply, but I'd appreciate for any thoughts from you!

**Comment by dmitridr** [comment]
Hi @donmen2 

As Mengying says, we don't support this right now, though I guess you could try editing the extension yourself to add this column in the feed. I can guide you through that if it interests you.

For the behavior you describe with the example of the skirt, this extension installs a pixel on your site, which will send back information to Facebook using the `id` column of the product that was viewed. If you are running `Dynamic Ads`, then Facebook then uses that information to show the person the skirt again, or something related to skirts, so I think there may be a way to do what you say without necessarily having the SKU.

**Comment by donmen2** [comment]
Hi,

I have created a php script that outputs exactly what I want/need, exactly the way I need. I have also created a cron job to regularly recreate that output. It's simple and efficient! I don't know if it's a safe script or not, but it does the job perfectly!

Thank you anyways!

**Comment by dmitridr** [comment]
Wow, glad to know that you solved this! Feel free to post the script here, I think it may help others in the same situation.
